lld/ELF/Writer.cpp

```diff
@@ -1464,6 +1466,12 @@ template <class ELFT> void Writer<ELFT>::finalizeAddressDependentContent() {
                     " does not converge");
         break;
       }
+    } else if (spilled) {
+      // Spilling can change relative section order, so recompute anything that
+      // depends on it.
+      for (Partition &part : partitions)
```

Comment:
Create a lambda to avoid duplicating the few lines. We might add more in the future and could accidentally break --enable-non-contiguous-regions